File: LayoutTests/http/tests/loading/preload-slow-loading.php
<?
header("Content-Type: text/html; charset=utf-8");
?>
<!-- <?php
# Spam a bunch of As to make sure we blow past any buffers.
print str_repeat("A", 2048);
?> -->
<script>
if (window.layoutTestController) {
    layoutTestController.dumpAsText();
    layoutTestController.dumpResourceResponseMIMETypes();
}
</script>
<script src="non-existant-1"></script>
<?php
flush();
sleep(1);
?>
<script>
document.write("<!--");
</script>
<img src="preload-slow-target.jpg">
<script>
document.write("-->");
</script>
<script src="non-existant-2"></script>

File: LayoutTests/http/tests/loading/preload-slow-loading.py
#!/usr/bin/env python3

import sys
import time

sys.stdout.write(
    'Content-Type: text/html; charset=utf-8\r\n\r\n'
    '<!-- {} -->\n'
    '<script>\n'
    'if (window.testRunner) {{\n'
    '    testRunner.dumpAsText();\n'
    '    testRunner.dumpResourceResponseMIMETypes();\n'
    '}}\n'
    '</script>\n'
    '<script src="non-existant-1"></script>\n'.format('A' * 2048)
)

sys.stdout.flush()
time.sleep(1)

sys.stdout.write(
    '<script>\n'
    'document.write("<!--");\n'
    '</script>\n'
    '<img src="preload-slow-target.jpg">\n'
    '<script>\n'
    'document.write("-->");\n'
    '</script>\n'
    '<script src="non-existant-2"></script>\n'
)